src/frontend/src/optimizer/rule/index_selection.rs

```diff
@@ -81,8 +90,7 @@ impl Rule for IndexSelectionRule {
             return None;
         }
 
-        let mut primary_table_scan_io_estimator = TableScanIoEstimator::new(logical_scan);
-        let primary_cost = primary_table_scan_io_estimator.estimate(logical_scan.predicate());
+        let primary_cost = self.estimate_table_scan_cost(logical_scan);
```

Comment:
actually I previously think the cost of scanning from primary should determined by index cost 🤔 .

Reply:
Primary table is also an index.